src/plugins/discover/public/application/components/doc_viewer/doc_viewer_render_tab.tsx

```diff
@@ -29,7 +29,7 @@
  */
 
 import React, { useRef, useEffect } from 'react';
-import { DocViewRenderFn, DocViewRenderProps } from '../../../doc_views/doc_views_types';
+import { DocViewRenderFn, DocViewRenderProps } from '../../doc_views/doc_views_types';
```

Comment:
How did this build if this wasn't imported correctly?

Reply:
I do see a type error when I debug the issue
```
Cannot find module '../../../doc_views/doc_views_types' or its corresponding type declarations.ts(2307)
```

I believe this is related to an old discussion https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1660, which highlights that the standard build scripts in the project do not fail on or show TypeScript errors. This has led to a situation where TypeScript compilation errors (like the one I encountered) do not cause the build to fail, allowing many such errors to creep into the project unnoticed.

Reply:
I believe we don't have the bandwidth to address this issue right now, but we should consider implementing a way to prevent new type errors from being added. This will help ensure the problem doesn't grow larger. 

CC: @AMoo-Miki @ashwin-pc

Reply:
Yeah typescript error right now rely on the IDE or us manually running tsc and validating the diff. I have a work in progress PR to help catch this in the CI but its not complete yet. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6586